test/trento_web/auth/jwt_auth_plug_test.exs

```diff
@@ -0,0 +1,218 @@
+defmodule TrentoWeb.JWTAuthPlugTest do
+  @moduledoc false
+
+  use TrentoWeb.ConnCase, async: true
+  use Plug.Test
+
+  alias TrentoWeb.Auth.{
+    AccessToken,
+    JWTAuthPlug,
+    RefreshToken
+  }
+
+  import Mox
+
+  @pow_config [otp_app: :trento]
+
+  setup [:set_mox_from_context, :verify_on_exit!]
+
+  describe "delete/2" do
+    test "should no-op and return the passed conn", %{conn: conn} do
+      res_conn = JWTAuthPlug.delete(conn, @pow_config)
+
+      assert conn == res_conn
+    end
+  end
+
+  describe "renew/2" do
+    test "should put in conn private a renewd access token if the refresh token is valid", %{
+      conn: conn
+    } do
+      expect(
+        Joken.CurrentTime.Mock,
+        :current_time,
+        8,
+        fn ->
+          1_671_715_992
+        end
+      )
+
+      valid_refresh = RefreshToken.generate_refresh_token!(%{"sub" => 1})
+
+      {:ok, res_conn} = JWTAuthPlug.renew(conn, valid_refresh)
+
+      assert %{
+               private: %{
+                 api_access_token: new_access_token,
+                 access_token_expiration: 600
+               }
+             } = res_conn
+
+      assert {:ok, %{"typ" => "Bearer"}} = Joken.peek_claims(new_access_token)
+    end
+
+    test "should return an error if the refresh token is malformed", %{conn: conn} do
+      expect(
+        Joken.CurrentTime.Mock,
+        :current_time,
+        0,
+        fn ->
+          1_671_715_992
+        end
+      )
+
+      {:error, _reason} = JWTAuthPlug.renew(conn, "invalid")
+    end
+
+    test "should return an error if the refresh token signature is invalid", %{conn: conn} do
+      expect(
+        Joken.CurrentTime.Mock,
+        :current_time,
+        0,
+        fn ->
+          1_671_715_992
+        end
+      )
+
+      bad_refresh =
```

Comment:
To make the test even more robust, can't these tokens be generated programmatically instead of using a magic string in the tests?

Reply:
The tokens need to be crafted accordingly, in order to set a proper time and date we will need to remock the current time and regenerate, or tweak the library to override the claims, imho it's better like that